### PR TITLE
Replace unittest with unittest2

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: tests/asyncunit.nim [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    template teardown(teardownBody) {.used.} =
+      teardown:
+        let asyncproc = proc {.async.} = teardownBody
+        waitFor asyncproc()
+
+    let suiteproc = proc = body # Avoids GcUnsafe2 warnings with chronos
+    suiteproc()
+
+template test*(name, body) =
```
Comment (arnetheduck) [inline tests/asyncunit.nim:29]: elsewhere, these have been dubbed `asyncTest` to avoid confusion (and import issues)

Comment (jangko) [reply tests/asyncunit.nim:29]: right, thank you.

**Comment by jlokier** [comment]
After this change, can test_rpc become part of all_tests without crashing?

**Comment by jangko** [comment]
it works now when I test it locally. let's see if CI tests agree with me.

**Pull request merged**